### PR TITLE
Bump ruby to 2.7.5

### PR DESCRIPTION
**Pull request by danilobuerger**
## Summary

This bumps ruby to the latest 2.7.x versions which includes bug fixes and 3 CVEs (https://www.ruby-lang.org/en/news/2021/11/24/ruby-2-7-5-released/)

## Changelog

[iOS] [Changed] - Bump ruby to 2.7.5

## Test Plan

no test plan, should just pass tests.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -94,7 +94,11 @@ commands:
     steps:
       - restore_cache:
           key: 1-gems-{{ checksum "Gemfile.lock" }}
-      - run: bundle check || bundle install --path vendor/bundle --clean
+      - run:
```
Comment (cortinico) [inline .circleci/config.yml:97]:
```suggestion
      - run:
```
There is a whitepsace here

**Review by cortinico — COMMENTED** [review]
Change looks good but I would hold-on on this till master is green again. @hramos is already aware of it and is looking into it.

Just want to make sure we don't break things even further

**Review by cortinico — APPROVED** [review]
This time the CI failure is a flaky test so we should be good to go

**Comment by danilobuerger** [comment]
@cortinico as @mikehardy mentioned in https://github.com/reactwg/react-native-releases/discussions/19#discussioncomment-2494872 it might be worthwhile to bump straight to the newest version of ruby instead of just the newest in the 2.7 release. What do you think? Are there any internal reasons to stay on 2.7?

**Comment by kelset** [comment]
I'm trying to learn if from the MSFT side of things we have some specific requirements about Ruby (for now, I couldn't find anything), but in the meantime I'm looking at the Ruby website:
* on the download page, they say current stable is 3.1.1 (https://www.ruby-lang.org/en/downloads/)
* on the maintenance branches one they say that 2.7, 3.0 and 3.1 are all on "normal maintenance" (https://www.ruby-lang.org/en/downloads/branches/)
So at this point in time it feels like it should be fine to move to 3.1.

But I'm still asking around MSFT internally to learn if there's any reason why we might want to stay on 2.7 so take what I'm saying with a pinch of salt

**Comment by cortinico** [comment]
> @cortinico as @mikehardy mentioned in [reactwg/react-native-releases#19 (reply in thread)](https://github.com/reactwg/react-native-releases/discussions/19#discussioncomment-2494872) it might be worthwhile to bump straight to the newest version of ruby instead of just the newest in the 2.7 release. What do you think? Are there any internal reasons to stay on 2.7?

I don't see any blockers on top of my mind. I believe we're good to go, as long as it concerns the iOS build pipeline. I would however suggest we:

1. Merge this PR first.
2. Do the bump to Ruby 3 in a separate PR as see if things are breaking because of that.

The reason is that it might not make it into 0.69.x and I believe we can anyway land this PR regardless 👍

**Pull request closed without merging**